ecommerce/frontend/src/context/AppContext.js

```diff
@@ -1,22 +1,91 @@
 import { notification } from 'antd'
 import constate from 'constate'
 import { useState } from 'react'
+import { api } from '../api'
 
 const guestUser = { type: 'guest' }
+
+// for testing purposes only
 const customerUser = { type: 'customer', name: 'Mehdi', surname: 'Saffar', email: 'example@gmail.com' }
 
 function useApp() {
-    const [user, setUser] = useState(customerUser)
+    const [user, setUser] = useState(guestUser)
+    const [requestInterceptorId, setRequestInterceptorId] = useState(null)
+
+    const requestInterceptor = request => {
+        request.headers.authorization = `Bearer ${localStorage.getItem('token')}`
+        return request
+    }
+
+    const init = async () => {
+        if (!localStorage.getItem('token')) {
+            console.log('no token in local storage, not attempting init')
+            return
+        }
+
+        try {
+            const { data } = await api.post('/init', null, {
+                headers: {
+                    authorization: localStorage.getItem('token'),
+                },
+            })
```

Comment:
you might also add 'Bearer ' prefix  to make post req. to /init ?

Reply:
Yes I added it in https://github.com/bounswe/bounswe2020group2/pull/185/commits/49ed54c60b33f70539ea726767fef6abd2b8cf77